code/modules/projectiles/guns/lever_action.dm

```diff
@@ -412,6 +412,58 @@ their unique feature is that a direct hit will buff your damage and firerate
 	recoil = RECOIL_AMOUNT_TIER_3
 	recoil_unwielded = RECOIL_AMOUNT_TIER_1
 
+/obj/item/weapon/gun/lever_action/xm88/wield(mob/user)
+	. = ..()
+
+	user.client?.mouse_pointer_icon = get_mouse_pointer(floating_penetration)
+	RegisterSignal(user, COMSIG_MOB_FIRED_GUN, PROC_REF(update_fired_mouse_pointer))
+
+/obj/item/weapon/gun/lever_action/xm88/unwield(mob/user)
+	. = ..()
+
+	user.client?.mouse_pointer_icon = null
+	UnregisterSignal(user, COMSIG_MOB_FIRED_GUN)
+
+/obj/item/weapon/gun/lever_action/xm88/proc/update_fired_mouse_pointer(mob/user)
+	SIGNAL_HANDLER
+
+	user.client?.mouse_pointer_icon = get_fired_mouse_pointer(floating_penetration)
+	addtimer(CALLBACK(src, PROC_REF(update_mouse_pointer), user), 0.4 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_CLIENT_TIME)
+
+/obj/item/weapon/gun/lever_action/xm88/proc/update_mouse_pointer(mob/user)
+	user.client?.mouse_pointer_icon = get_mouse_pointer(floating_penetration)
+
+/obj/item/weapon/gun/lever_action/xm88/proc/get_mouse_pointer(level)
```

Comment:
can't both these procs be converted to `return 'icons/effects/mouse_pointer/xm88/xm88-[level].dmi'`?

Reply:
god i wish - ' ' can't have expressions in them, and file("icons/whatever/xm88-[level]") also... does not work. mouse_pointer_icon is a truly cursed

Reply:
> god i wish - ' ' can't have expressions in them, and file("icons/whatever/xm88-[level]") also... does not work. mouse_pointer_icon is a truly cursed var

How about `icon("icons/whatever/xm88-[level]")`?

Reply:
this also doesn't work 😭